internal/clients/client.go

```diff
@@ -114,6 +114,9 @@ func (cl *Client) UpdateSourceChannel(d *schema.ResourceData) (*Client, error) {
 
 		// Return a new client with the updated source channel
 		cl, err = NewClient(cl.Config)
+		if err != nil {
+			return cl, fmt.Errorf("failed to create new client with updated source channel: %v", err)
```

Comment:
I don't think we want to throw an error here; just a warning is enough. On the slim chance there's an actual error, it only means a few requests will have user agent headers missing HCP provider version info. Not worth failing the provider for.